src/Alg/Signature/SignatureAlgorithmFactory.php

```diff
@@ -4,32 +4,30 @@
 
 namespace SimpleSAML\XMLSecurity\Alg\Signature;
 
-use SimpleSAML\Assert\Assert;
-use SimpleSAML\XMLSecurity\Alg\SignatureAlgorithm;
+use SimpleSAML\XMLSecurity\Alg\AbstractAlgorithmFactory;
 use SimpleSAML\XMLSecurity\Constants as C;
-use SimpleSAML\XMLSecurity\Exception\BlacklistedAlgorithmException;
-use SimpleSAML\XMLSecurity\Exception\InvalidArgumentException;
-use SimpleSAML\XMLSecurity\Exception\UnsupportedAlgorithmException;
 use SimpleSAML\XMLSecurity\Key\AbstractKey;
 
-use function in_array;
-
 /**
  * Factory class to create and configure digital signature algorithms.
  *
  * @package simplesamlphp/xml-security
  */
-final class SignatureAlgorithmFactory
+final class SignatureAlgorithmFactory extends AbstractAlgorithmFactory
 {
     /**
-     * An array holding the known classes extending \SimpleSAML\XMLSecurity\Alg\Signature\AbstractSigner.
+     * A cache of algorithm implementations indexed by algorithm ID.
      *
      * @var string[]
      */
-    private static array $algorithms = [
-        RSA::class,
-        HMAC::class
-    ];
+    protected static array $cache = [];
```

Comment:
Why the change from private to protected? The class is final anyway.

Reply:
It needs to be protected to be accessible from the abstract parent class.